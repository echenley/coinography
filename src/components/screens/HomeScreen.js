// @flow

import React from 'react'
import { Button, Card, Container, Content, Right, Text } from 'native-base'
import type { NavigationNavigatorProps } from 'react-navigation'
import Chart from '../Chart'
import Controls from '../Controls'

class HomeScreen extends React.Component<{}> {
  static navigationOptions = ({
    navigation,
  }: NavigationNavigatorProps<*, *>) => ({
    title: 'Coinography',
    headerRight: (
      <Right>
        <Button onPress={() => navigation.navigate('About')} transparent>
          <Text>About</Text>
        </Button>
      </Right>
    ),
  })

  render() {
    return (
      <Container>
        <Content padder>
          <Card>
            <Controls />
            <Chart />
          </Card>
        </Content>
      </Container>
    )
  }
}

export default HomeScreen
