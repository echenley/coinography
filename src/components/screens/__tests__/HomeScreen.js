import * as React from 'react'
import { shallow } from 'enzyme'
import HomeScreen from '../HomeScreen'

const renderComponent = () => shallow(<HomeScreen />)

describe('<HomeScreen />', () => {
  it('renders correctly', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('has correct navigation options', () => {
    const navigation = {
      navigate: jest.fn(),
    }

    const options = HomeScreen.navigationOptions({ navigation })

    expect(options.title).toBe('Coinography')
    expect(options.headerRight).toMatchSnapshot('headerRight')

    options.headerRight.props.children.props.onPress()
    expect(navigation.navigate).toHaveBeenCalledWith('About')
  })
})
