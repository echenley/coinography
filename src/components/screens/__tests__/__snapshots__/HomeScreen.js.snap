// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> has correct navigation options: headerRight 1`] = `
<Styled(Right)
  virtual={undefined}
>
  <Styled(Button)
    onPress={[Function]}
    transparent={true}
    virtual={undefined}
  >
    <Styled(Text)
      virtual={undefined}
    >
      About
    </Styled(Text)>
  </Styled(Button)>
</Styled(Right)>
`;

exports[`<HomeScreen /> renders correctly 1`] = `
<Styled(Container)
  virtual={undefined}
>
  <Styled(Content)
    padder={true}
    virtual={undefined}
  >
    <Styled(Card)
      virtual={undefined}
    >
      <Connect(Controls) />
      <Connect(Chart) />
    </Styled(Card)>
  </Styled(Content)>
</Styled(Container)>
`;
